climpyrical/data.py

```diff
@@ -80,30 +84,53 @@ def read_data(data_path: str, required_keys: list = None) -> xr.Dataset:
             "climpyrical requires a NetCDF4 file with extension .nc"
         )
 
-    if required_keys is None:
-        required_keys = ["rlat", "rlon"]
-
     with xr.open_dataset(data_path) as ds:
         all_keys = list(set(ds.variables).union(set(ds.dims)))
+
         check_valid_keys(all_keys, required_keys)
         check_valid_data(ds)
 
-        if "time" in ds.keys() and ds["time"].size <= 1:
-            ds = ds.squeeze("time").drop_vars("time")
-        if "time_bnds" in ds.keys():
-            ds = ds.drop_vars("time_bnds")
-        if "rotated_pole" in ds.keys():
-            ds = ds.drop_vars("rotated_pole")
+        # keys with size > 2 are good, otherwise
+        # are superfluous (time, time_bnds, rotated_pole, etc)
+        extra_keys = [key for key in all_keys if ds[key].size <= 2]
+
+        # check how many data variables with
```

Comment:
[In `nchelpers`, we call these "dependent variables"](https://github.com/pacificclimate/nchelpers/blob/master/nchelpers/__init__.py#L948). Have you tried using `nchelpers` for any of this work?

Reply:
No, but I'm keen to take a look. :smile:

Reply:
A note on this: I didn't end up using `nchelpers` mainly because I stick with `xarray.Dataset` objects throughout. I did however, adopt a very similar approach to what's in [dependent variables](https://github.com/pacificclimate/nchelpers/blob/master/nchelpers/__init__.py#L948), instead using `xarray.Dataset`s

i.e.

```python
        for variable in self.variables.values():
            if hasattr(variable, 'bounds'):
                non_dependent_var_names.add(variable.bounds)
            if hasattr(variable, 'climatology'):
                non_dependent_var_names.add(variable.climatology)
            if hasattr(variable, 'grid_mapping'):
                non_dependent_var_names.add(variable.grid_mapping)
            if hasattr(variable, 'coordinates'):
                non_dependent_var_names.update(variable.coordinates.split())
```

Isn't very helpful for me since those attributes don't really show up. `read_data` is able to detect these non dependent variables without explicitly checking them. This PR is an improvement on [what was done in the past](https://github.com/pacificclimate/climpyrical/blob/0e407c9789bce4cc3c9045989c438311175f287e/climpyrical/data.py#L91)